### PR TITLE
unify wave/wave-nompi

### PR DESCRIPTION
**Pull request by matthiasdiener**
somewhat similar to #553.

**Questions for the review**:
- [ ] Is the scope and purpose of the PR clear?
  - [ ] The PR should have a description.
  - [ ] The PR should have a guide if needed (e.g., an ordering).
- [ ] Is every top-level method and class documented? Are things that should be documented actually so?
- [ ] Is the interface understandable? (I.e. can someone figure out what stuff does?) Is it well-defined?
- [ ] Does the implementation do what the docstring claims?
- [ ] Is everything that is implemented covered by tests?
- [ ] Do you see any immediate risks or performance disadvantages with the design? Example: what do interface normals attach to?

### CODE REVIEW

File: examples/wave.py [hunk 1]
```diff
@@ -256,14 +275,23 @@ def rhs(t, w):
...
 
-    main(actx_class, use_logmgr=args.log, casename=casename)
+    if args.nompi:
+        import inspect
+        # run main without the mpi_entry_point wrapper
+        inspect.unwrap(main)(
+            actx_class, use_logmgr=args.log, casename=casename,
+            mpi=not args.nompi)
+    else:
+        main(actx_class, use_logmgr=args.log, casename=casename, mpi=not args.nompi)
```
Comment (majosm) [inline examples/wave.py:295]: I think you can do this with:

```suggestion
    if args.mpi:
        main_func = mpi_entry_point(main)
    else:
        main_func = main

    main_func(actx_class, use_logmgr=args.log, casename=casename, mpi=args.mpi)
```

(and removing the decorator on `main`).

File: examples/run_examples.sh [hunk 2]
```diff
@@ -126,11 +120,7 @@ do
         basic_command="python -m mpi4py ${example_path} --casename ${test_name}"
         [[ $actx != "eager" ]] && basic_command+=" --$actx"
         set -x
-        if [[ "$nompi" -gt 0 ]]; then
-            ${basic_command}
-        else
-            ${mpi_exec} -n 2 $mpi_launcher $basic_command
-        fi
+        ${mpi_exec} -n 2 $mpi_launcher $basic_command
```
[the remark this answers is not in the document]
Comment (matthiasdiener) [reply examples/run_examples.sh:123]: Yes, correct. No need I think, the only reason to run a non-MPI driver is to debug MPI-related issues, which are super rare.

File: examples/wave.py [hunk 3]
```diff
@@ -85,27 +92,38 @@ def main(actx_class, casename="wave",
...
+        # Distribute mesh
+        if comm:
+            if mesh_dist.is_mananger_rank():
+                from meshmode.distributed import get_partition_by_pymetis
+                part_per_element = get_partition_by_pymetis(mesh, num_parts)
+
+                local_mesh = mesh_dist.send_mesh_parts(
+                    mesh, part_per_element, num_parts)
+
+                del mesh
```
[the remark this answers is not in the document]
Comment (matthiasdiener) [reply examples/wave.py]: What do you think of https://github.com/illinois-ceesd/mirgecom/pull/969/commits/40e26d61092e62fa7476324fb7a84ff85fab61f5?

**Review by majosm — APPROVED** [review]
👍

**Pull request merged**